### PR TITLE
Transaction BBE update

### PR DESCRIPTION
**Pull request by rdhananjaya**
Transaction BBE update with new syntax.

### CODE REVIEW

File: examples/local-transactions-with-participants/local_transaction_participants.description [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// A Ballerina transaction is a series of data manipulation statements 
+// that must either fully complete or fully fail, thereby leaving the
+// system in a consistent state. This sample demonstrate usage of transaction
```
Comment (Manuri) [inline examples/local-transactions-with-participants/local_transaction_participants.description]:
```suggestion
// system in a consistent state. This sample demonstrates usage of transaction
```

File: examples/local-transactions-with-participants/local_transaction_participants.description [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+// local participants. If participant panics, even when it is trapped and
+// initiator completes, the the transaction is considered failure.
```
Comment (Manuri) [inline examples/local-transactions-with-participants/local_transaction_participants.description]:
```suggestion
// initiator completes, then the transaction is considered failure.
```

File: examples/local-transactions-with-participants/local_transaction_participants.bal [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import ballerina/io;
+// Import transactions moudle in order to use participant annotation.
```
Comment (anupama-pathirage) [inline examples/local-transactions-with-participants/local_transaction_participants.bal]: typo moudle

File: examples/local-transactions-with-participants/local_transaction_participants.bal [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import ballerina/transactions;
```
Comment (anupama-pathirage) [inline examples/local-transactions-with-participants/local_transaction_participants.bal]: Why do we need to import transactions package here?

Comment (anupama-pathirage) [reply examples/local-transactions-with-participants/local_transaction_participants.bal]: Sorry for the noise. Just now noticed the annotation.

Comment (rdhananjaya) [reply examples/local-transactions-with-participants/local_transaction_participants.bal]: It's for `@transactions:Participant` annotation

File: examples/local-transactions/local_transactions.bal [hunk 5]
```diff
@@ -27,31 +27,25 @@ public function main() {
...
-                                  onabort = onAbortFunction {
-        // This is the first remote function participant in the transaction.
-        var result = testDB->update("INSERT INTO CUSTOMER(ID,NAME)
+    transaction with retries = 4 {
+    // This is the first remote function participant in the transaction.
+        var count = testDB->update("INSERT INTO CUSTOMER(ID,NAME)
                                      VALUES (1, 'Anne')");
-        // This is the second remote function participant in the transaction.
-        result = testDB->update("INSERT INTO SALARY (ID, MON_SALARY)
+        // This is the second action participant in the transaction.
```
Comment (anupama-pathirage) [inline examples/local-transactions/local_transactions.bal]: Action should be changed to remote function

File: examples/xa-transactions/xa_transactions.bal [hunk 6]
```diff
@@ -89,8 +90,9 @@ function onAbortFunction(string transactionId) {
...
-    match returned {
-        int retInt => io:println(message + " status: " + retInt);
-        error err => io:println(message + " failed: " + err.message);
+    if (returned is int) {
+        io:println(message + " status: " + returned);
+    } else {
+        io:println(message + " failed: " + returned.reason());
     }
-}
+}
```
Comment (anupama-pathirage) [inline examples/xa-transactions/xa_transactions.bal]: Add a new line here

File: examples/xa-transactions/xa_transactions.out [hunk 7]
```diff
@@ -1,16 +1,15 @@
...
 Generated key for the inserted row: 1
 Insert to SALARY table status: 1
-2018-10-30 10:20:49,392 INFO  [ballerina/transactions] - Running 2-phase commit for transaction: 15d1e705-62f4-4b9b-80d3-7a35d5262266:1
-Transaction: 15d1e705-62f4-4b9b-80d3-7a35d5262266:1 committed
+2018-12-04 20:12:03,541 INFO  [ballerina/log] - Running 2-phase commit for transaction: aec7b091-cbac-4d61-8834-bed8e1a3c9ec:$anon$.$1
+Transaction committed
 Drop Table CUSTOMER status: 0
-Drop Table SALARY status: 0
-[ballerina/http] stopped HTTP/WS endpoint 172.17.0.1:13827
+Drop Table SALARY status: 0
```
Comment (anupama-pathirage) [inline examples/xa-transactions/xa_transactions.out]: Add new line here

**Pull request merged**